starlite/middleware/compression/brotli.py

```diff
@@ -0,0 +1,147 @@
+import io
+from typing import Union
+
+from starlette.datastructures import Headers, MutableHeaders
+from starlette.middleware.gzip import GZipResponder, unattached_send
+from starlette.types import ASGIApp, Message, Receive, Scope, Send
+
+from starlite.config import BrotliMode
+from starlite.enums import ScopeType
+from starlite.exceptions import MissingDependencyException
+
+try:
+    import brotli
+except ImportError as exc:  # pragma: no cover
+    raise MissingDependencyException("brotli is not installed") from exc
+
+
+class BrotliMiddleware:
+    def __init__(
+        self,
+        app: ASGIApp,
+        minimum_size: int = 400,
+        brotli_quality: int = 4,
+        brotli_mode: BrotliMode = BrotliMode.TEXT,
+        brotli_lgwin: int = 22,
+        brotli_lgblock: int = 0,
+        brotli_gzip_fallback: bool = True,
+    ) -> None:
+        self.app = app
+        self.quality = brotli_quality
+        self.mode = _brotli_mode_lookup(brotli_mode)
+        self.minimum_size = minimum_size
+        self.lgwin = brotli_lgwin
+        self.lgblock = brotli_lgblock
+        self.gzip_fallback = brotli_gzip_fallback
+
+    async def __call__(self, scope: Scope, receive: Receive, send: Send) -> None:
+        if scope["type"] == ScopeType.HTTP:
+            headers = Headers(scope=scope)
+            if "br" in headers.get("Accept-Encoding", ""):
+                responder = BrotliResponder(
+                    app=self.app,
+                    minimum_size=self.minimum_size,
+                    quality=self.quality,
+                    mode=self.mode,
+                    lgwin=self.lgwin,
+                    lgblock=self.lgblock,
+                )
+                await responder(scope, receive, send)
+                return
+            elif self.gzip_fallback and "gzip" in headers.get("Accept-Encoding", ""):
```

Comment:
no need for elif since you return above.
```suggestion
            if self.gzip_fallback and "gzip" in headers.get("Accept-Encoding", ""):
```